src/libasr/asr_verify.cpp

```diff
@@ -279,6 +280,14 @@ class VerifyVisitor : public BaseWalkVisitor<VerifyVisitor>
         current_symtab = parent_symtab;
     }
 
+    void visit_DerivedType(const DerivedType_t& x) {
+        visit_DerivedTypeEnumType(x);
+    }
+
+    void visit_EnumType(const EnumType_t& x) {
+        visit_DerivedTypeEnumType(x);
+    }
```

Comment:
Let's add more checks here. 

* We require all members of `Enum` to have the `value` (?) to be set.
* We require all `value` to be of the same type, and that type to be equal to the `type` in `Enum`.

I think we do not have any other restrictions at this point. Later we will.

Reply:
> * We require all `value` to be of the same type, and that type to be equal to the `type` in `Enum`.

This is already done in AST to ASR transition as we cannot create EnumType without this condition being satisfied. But anyways I will add the checks here as well.

> We require all members of Enum to have the value (?) to be set.

Correct. Will add the checks.

Reply:
Done.

Reply:
Thanks. The idea of verify is that it checks all conditions that we require ASR to satisfy. Of course, the idea is that the frontend satisfies them all, so that in release mode we do not need to run verify, for performance reasons. But for development and debugging we want to do that.